app/activity/[id]/components/activity-info.tsx

```diff
@@ -1,31 +1,32 @@
 'use client'
 
 import { Heading, Box, Text, VStack } from '@chakra-ui/react'
-
-type ActivityData = {
-  memo: string
-  cost: string
-}
+import { ActivityCollectionQuery } from '@generated/api'
 
 type ActivityInfoProps = {
-  dummyActivityData: ActivityData
+  activityData: ActivityCollectionQuery | undefined
 }
 
-export const ActivityInfo = ({ dummyActivityData }: ActivityInfoProps) => {
+export const ActivityInfo: React.FC<ActivityInfoProps> = ({ activityData }) => {
+  if (!activityData || !activityData.activityCollection) {
+    return null
+  }
+
+  const data = activityData.activityCollection.edges[0].node
```

Comment:
I'd say we don't have to specify this condition due to the reason that I mentioned in the other part.